hikari/utilities/net.py

```diff
@@ -39,18 +39,24 @@ async def generate_error_response(response: aiohttp.ClientResponse) -> errors.HT
     real_url = str(response.real_url)
     raw_body = await response.read()
 
+    cls: typing.Optional[typing.Type[errors.HTTPError]] = None
+
     if response.status == http.HTTPStatus.BAD_REQUEST:
-        return errors.BadRequestError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.UNAUTHORIZED:
-        return errors.UnauthorizedError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.FORBIDDEN:
-        return errors.ForbiddenError(real_url, response.headers, raw_body)
-    if response.status == http.HTTPStatus.NOT_FOUND:
-        return errors.NotFoundError(real_url, response.headers, raw_body)
+        cls = errors.BadRequestError
+    elif response.status == http.HTTPStatus.UNAUTHORIZED:
+        cls = errors.UnauthorizedError
+    elif response.status == http.HTTPStatus.FORBIDDEN:
+        cls = errors.ForbiddenError
+    elif response.status == http.HTTPStatus.NOT_FOUND:
+        cls = errors.NotFoundError
+
+    if cls is not None:
+        json_body = await response.json()
+        # False positive
+        return cls(real_url, response.headers, raw_body, json_body["message"], json_body["code"])  # type: ignore[call-arg]
```

Comment:
which rest api version are we targetting here? I'm not sure about v7 and v8 but I know for sure that not all sane error response will have "message" and "code" in their body on v6 (a good example of this is when you try to pass an invalid embed on create message or execute webhook)

Reply:
The docs state that all the errors will include a `code` and `message` along with the status. Will have a look at those exceptions you said

Reply:
The docs sate "an also return more detailed error codes" where emphasis should be put on the usage of "can" as opposed to "will", there's no real guarantee there

Reply:
Ok, will change so that it doesnt assume that they send `message` nor `code` and will also fix the other thing you pointed out with cloudfare